File: backend/check-users-structure.js
const { Pool } = require('pg');

const pool = new Pool({
  host: 'localhost',
  port: 5432,
  user: 'eff_admin',
  password: 'Frames!123',
  database: 'eff_membership_database'
});

async function checkUsersStructure() {
  try {
    console.log('🔍 Checking users table structure...\n');
    
    // Get table columns
    const columns = await pool.query(`
      SELECT column_name, data_type, is_nullable
      FROM information_schema.columns
      WHERE table_schema = 'public'
        AND table_name = 'users'
      ORDER BY ordinal_position
    `);
    
    console.log('users table columns:');
    console.table(columns.rows);
    
    // Get user count
    const userCount = await pool.query('SELECT COUNT(*) as count FROM users');
    console.log('\nTotal users:', userCount.rows[0].count);
    
    // Get first 5 users (using only user_id)
    const firstUsers = await pool.query('SELECT * FROM users ORDER BY user_id LIMIT 5');
    console.log('\nFirst 5 users:');
    console.table(firstUsers.rows);
    
    // Check if user_id 1 exists
    const user1 = await pool.query('SELECT * FROM users WHERE user_id = 1');
    console.log('\nUser ID 1 exists:', user1.rows.length > 0);
    if (user1.rows.length > 0) {
      console.log('User ID 1 details:');
      console.table(user1.rows);
    }
    
    // Check the foreign key constraint
    console.log('\n🔍 Checking foreign key constraint on leadership_appointments...\n');
    const fkConstraint = await pool.query(`
      SELECT
        tc.constraint_name,
        tc.table_name,
        kcu.column_name,
        ccu.table_name AS foreign_table_name,
        ccu.column_name AS foreign_column_name
      FROM information_schema.table_constraints AS tc
      JOIN information_schema.key_column_usage AS kcu
        ON tc.constraint_name = kcu.constraint_name
        AND tc.table_schema = kcu.table_schema
      JOIN information_schema.constraint_column_usage AS ccu
        ON ccu.constraint_name = tc.constraint_name
        AND ccu.table_schema = tc.table_schema
      WHERE tc.constraint_type = 'FOREIGN KEY'
        AND tc.table_name = 'leadership_appointments'
        AND kcu.column_name IN ('appointed_by', 'terminated_by')
    `);
    
    console.log('Foreign key constraints on appointed_by and terminated_by:');
    console.table(fkConstraint.rows);
    
  } catch (error) {
    console.error('❌ Error:', error.message);
  } finally {
    await pool.end();
  }
}

checkUsersStructure();

File: test/check_users_structure.py
#!/usr/bin/env python3
"""
Check users table structure and related tables
"""

import psycopg2

conn = psycopg2.connect(
    host='localhost',
    user='eff_admin',
    password='Frames!123',
    database='eff_membership_database',
    port=5432
)
cur = conn.cursor()

print("=" * 100)
print("USERS TABLE STRUCTURE")
print("=" * 100)

# Get users table columns
cur.execute("""
    SELECT 
        column_name, 
        data_type, 
        character_maximum_length,
        is_nullable,
        column_default
    FROM information_schema.columns 
    WHERE table_name = 'users'
    ORDER BY ordinal_position
""")

print("\nColumns in users table:")
print("-" * 100)
for col in cur.fetchall():
    nullable = "NULL" if col[3] == 'YES' else "NOT NULL"
    max_len = f"({col[2]})" if col[2] else ""
    default = f" DEFAULT {col[4]}" if col[4] else ""
    print(f"  {col[0]:<30} {col[1]}{max_len:<20} {nullable:<10} {default}")

# Check roles table
print("\n" + "=" * 100)
print("ROLES TABLE")
print("=" * 100)

cur.execute("SELECT role_id, role_name, description FROM roles ORDER BY role_id")
roles = cur.fetchall()
print("\nAvailable roles:")
print("-" * 100)
for role in roles:
    print(f"  {role[0]}: {role[1]:<20} - {role[2]}")

# Check current users
print("\n" + "=" * 100)
print("CURRENT USERS IN DATABASE")
print("=" * 100)

cur.execute("""
    SELECT 
        u.user_id,
        u.name,
        u.email,
        u.role_id,
        r.role_name,
        u.admin_level,
        u.member_id,
        u.is_active
    FROM users u
    LEFT JOIN roles r ON u.role_id = r.role_id
    ORDER BY u.user_id
    LIMIT 10
""")

users = cur.fetchall()
print(f"\nShowing first 10 users (Total: {len(users)}):")
print("-" * 100)
for user in users:
    print(f"  ID: {user[0]:<6} | {user[1]:<30} | {user[2]:<35} | Role: {user[4]:<15} | Level: {user[5]:<10} | Active: {user[7]}")

# Check geographic hierarchy tables
print("\n" + "=" * 100)
print("GEOGRAPHIC HIERARCHY")
print("=" * 100)

cur.execute("SELECT COUNT(*) FROM provinces")
print(f"  Provinces: {cur.fetchone()[0]}")

cur.execute("SELECT COUNT(*) FROM districts")
print(f"  Districts: {cur.fetchone()[0]}")

cur.execute("SELECT COUNT(*) FROM municipalities")
print(f"  Municipalities: {cur.fetchone()[0]}")

cur.execute("SELECT COUNT(*) FROM wards")
print(f"  Wards: {cur.fetchone()[0]}")

conn.close()

print("\n" + "=" * 100)
print("END OF REPORT")
print("=" * 100)

